docs/smart-contracts/guides/transactions/simulateTransaction-Deep-Dive.mdx

```diff
@@ -0,0 +1,531 @@
+    return false;
+  }
+
+  switch (tx.operations[0].type) {
+    case 'invokeHostFunction':
+    case 'extendFootprintTtl':
+    case 'restoreFootprint':
+      return true;
+
+    default:
+      return false;
+  }
+}
+```
+
+## Handling Archived Ledger Entries
+
+When a ledger entry is archived, it needs to be restored before the transaction can be submitted. This is indicated in the `restorePreamble` field of the result.
+
+```typescript
+interface RestorePreamble {
+  minResourceFee: string; // Recommended minimum resource fee to add when submitting the RestoreFootprint operation.
+  transactionData: string; // The recommended Soroban Transaction Data to use when submitting the RestoreFootprint operation.
+}
+```
+
+Here is an example for handling restoration using the `restorePreamble` to restore archived data:
+
+```typescript
+/**
+ * Simulates a restoration transaction to determine if restoration is needed.
+ * This function first checks the ledger entry for the given WASM hash. If the entry is found and has expired,
+ * it attempts a restoration. If the entry hasn't expired yet but the TTL needs extension, it proceeds with TTL extension.
+ * @param wasmHash - The hash of the WASM to check.
+ * @param txParams - The transaction parameters including account and signer.
+ * @returns A promise that resolves to a simulation response, indicating whether restoration or TTL extension is needed.
+ */
+export async function simulateRestorationIfNeeded(
+  wasmHash: Buffer,
+  txParams: TxParams,
+): Promise<
+  SorobanRpc.Api.SimulateTransactionRestoreResponse | string | undefined
+> {
+  try {
+    const account = await config.rpc.getAccount(txParams.account.accountId());
+    const ledgerKey = getLedgerKeyWasmId(wasmHash);
+    const response = await config.rpc.getLedgerEntries(...[ledgerKey]);
+    if (
+      response.entries &&
+      response.entries.length > 0 &&
+      response.entries[0].liveUntilLedgerSeq
+    ) {
+      const expirationLedger = response.entries[0].liveUntilLedgerSeq;
+      const desiredLedgerSeq = response.latestLedger + 500000;
+      let extendLedgers = desiredLedgerSeq - expirationLedger;
+      if (extendLedgers < 10000) {
+        extendLedgers = 10000;
+      }
+      console.log("Expiration:", expirationLedger);
+      console.log("Desired TTL:", desiredLedgerSeq);
+      const sorobanData = new SorobanDataBuilder()
+        .setReadWrite([ledgerKey])
+        .build();
+      const restoreTx = new TransactionBuilder(
+        account,
+        txParams.txBuilderOptions,
+      )
+        .setSorobanData(sorobanData)
+        .addOperation(Operation.restoreFootprint({})) // The actual restore operation
+        .build();
+      // Simulate a transaction with a restoration operation to check if it's necessary
+
+      const restorationSimulation: SorobanRpc.Api.SimulateTransactionResponse =
+        await config.rpc.simulateTransaction(restoreTx);
+      const restoreNeeded = SorobanRpc.Api.isSimulationRestore(
```

Comment:
restore is only needed if it succeeds, right? so that one should be checked first

Reply:
ah yeah i pulled that code from a larger example i had called something like simulateIfRestorationNeeded, I will review that and update. Thanks!

Reply:
Are you implying i only need to check isSimulationSuccess() here?

Reply:
Yeah I'm just saying that you should check for success and then check for restoration in your conditional because restore implies success (ref: [sdk code](https://github.com/stellar/js-stellar-sdk/blob/master/src/rpc/api.ts#L319)).

Reply:
> Yeah I'm just saying that you should check for success and then check for restoration in your conditional because restore implies success (ref: [sdk code](https://github.com/stellar/js-stellar-sdk/blob/master/src/rpc/api.ts#L319)).

i'm changing it to only check for issimulationrestore since the sdk code you referenced shows that already checks for success.
```ts
 export function isSimulationRestore(
    sim: SimulateTransactionResponse
  ): sim is SimulateTransactionRestoreResponse {
    return (
      isSimulationSuccess(sim) &&
      'restorePreamble' in sim &&
      !!sim.restorePreamble.transactionData
    );
  }
  ```